rust/src/storage/s3/mod.rs

```diff
@@ -416,13 +417,47 @@ impl StorageBackend for S3StorageBackend {
         debug!("delete s3 object: {}...", path);
 
         let uri = parse_uri(path)?.into_s3object()?;
-        let put_req = DeleteObjectRequest {
+        let delete_req = DeleteObjectRequest {
             bucket: uri.bucket.to_string(),
             key: uri.key.to_string(),
             ..Default::default()
         };
 
-        self.client.delete_object(put_req).await?;
+        self.client.delete_object(delete_req).await?;
+
+        Ok(())
+    }
+
+    async fn delete_objs(&self, paths: &[String]) -> Result<(), StorageError> {
+        debug!("delete s3 objects: {:?}...", paths);
+
+        let uris = paths
+            .iter()
+            .map(|path| {
+                let s3_obj = parse_uri(path).unwrap().into_s3object().unwrap();
+                (s3_obj.bucket, s3_obj)
+            })
+            .into_group_map();
+
+        for (bucket, objs) in uris {
```

Comment:
This still does only one at a time. 
We should chunk them with maximum 1000 at once
 https://github.com/delta-io/delta-rs/pull/396/files#diff-87e43fff59e072c5dbd99295f15fbf86d5a0e53f6586689191b2c61e07ace91aR439
 
 and also checking whether the buckets are the same for all paths makes sense I think.

Reply:
Yes, I think it should be the same bucket in vacuum operation.